Composer/packages/client/src/store/reducer/index.ts

```diff
@@ -97,44 +107,42 @@ const removeRecentProject: ReducerFunc = (state, { path }) => {
   return state;
 };
 
-const updateDialog: ReducerFunc = (state, { id, content }) => {
-  state.dialogs = state.dialogs.map(dialog => {
-    if (dialog.id === id) {
-      const result = dialogIndexer.parse(dialog.id, content, state.schemas.sdk.content);
-      return { ...dialog, ...result };
-    }
-    return dialog;
-  });
-  return state;
-};
-
-const removeDialog: ReducerFunc = (state, { response }) => {
-  state.dialogs = response.data.dialogs;
-  state.luFiles = updateLuFilesStatus(state.botName, response.data.luFiles);
-  state.lgFiles = response.data.lgFiles;
-  return state;
-};
-
-const createDialogBegin: ReducerFunc = (state, { actionsSeed, onComplete }) => {
-  state.showCreateDialogModal = true;
-  state.actionsSeed = actionsSeed;
-  state.onCreateDialogComplete = onComplete;
-  return state;
-};
+const createLgFile: ReducerFunc = (state, { id, content }) => {
+  const { lgFiles, locale } = state;
+  id = `${id}.${locale}`;
+  if (lgFiles.find(lg => lg.id === id)) {
+    state.error = {
+      message: `${id} ${formatMessage(`lg file already exist`)}`,
+      summary: formatMessage('Creation Rejected'),
+    };
+    return state;
+  }
+  // slot with common.lg import
+  let lgInitialContent = '';
+  const lgCommonFile = lgFiles.find(({ id }) => id === `common.${locale}`);
+  if (lgCommonFile) {
+    lgInitialContent = `[import](common.lg)`;
+  }
+  content = [lgInitialContent, content].join('\n');
 
-const createDialogCancel: ReducerFunc = state => {
-  state.showCreateDialogModal = false;
-  delete state.onCreateDialogComplete;
+  const { parse } = lgIndexer;
+  const lgImportresolver = importResolverGenerator(state.lgFiles, '.lg');
+  const { templates, diagnostics } = parse(content, id, lgImportresolver);
+  const lgFile = { id, templates, diagnostics, content };
+  state.lgFiles.push(lgFile);
+  filePersistence.notify(FileChangeType.CREATE, id, FileExtensions.Lg, content);
   return state;
 };
 
-const createDialogSuccess: ReducerFunc = (state, { response }) => {
-  state.dialogs = response.data.dialogs;
-  state.luFiles = updateLuFilesStatus(state.botName, response.data.luFiles);
-  state.lgFiles = response.data.lgFiles;
-  state.showCreateDialogModal = false;
-  state.actionsSeed = [];
-  delete state.onCreateDialogComplete;
+const removeLgFile: ReducerFunc = (state, { id }) => {
+  state.lgFiles = state.lgFiles.reduce((result: LgFile[], file) => {
+    if (getBaseName(file.id) === id || file.id === id) {
+      filePersistence.notify(FileChangeType.DELETE, file.id, FileExtensions.Lg, '');
```

Comment:
We should not require each reducer to manually add a notify. The notification should be done by your framework.